packages/core/content-manager/admin/src/preview/pages/Preview.tsx

```diff
@@ -116,6 +144,23 @@ const PreviewPage = () => {
   );
   const device = DEVICES.find((d) => d.name === deviceName) ?? DEVICES[0];
 
+  // Listen for ready message from iframe before injecting script
+  React.useEffect(() => {
+    const handleMessage = (event: MessageEvent) => {
+      if (event.data?.type === PUBLIC_EVENTS.PREVIEW_READY) {
+        const script = `(${previewScript.toString()})()`;
```

Comment:
Just confused what's happening here. You stringify the script but then what's the `()` about? Does it become a callable function?

Reply:
When I stringify the function, what I get is its internal implementation. I could just dump it as the script content, but then it would require top-level await support, which doesn't have as good compatibility as regular await because it's more recent.

So it's safer to wrap it in a function and call it, which is what I'm doing here via a self invoking function. The long form version would be this, I can change it it you prefer:

```
const script = `
  const runScript = () => {
    ${previewScript.toString()}
  }
  runScript()
```